scaffoldmaker/meshtypes/meshtype_3d_colon1.py

```diff
@@ -0,0 +1,141 @@
+from scaffoldmaker.meshtypes.scaffold_base import Scaffold_base
+from scaffoldmaker.utils.matrix import *
+from scaffoldmaker.utils.meshrefinement import MeshRefinement
+from scaffoldmaker.utils.tubemesh import *
+
+class MeshType_3d_colon1(Scaffold_base):
+    '''
+    Generates a 3-D colon mesh with variable numbers
+    of elements around, along the central line, and through wall.
+    The colon is created by a function that generates a haustra
+    segment and uses tubemesh to map the segment along a central
+    line profile.
+    '''
+    @staticmethod
+    def getName():
+        return '3D Colon 1'
+
+    @staticmethod
+    def getParameterSetNames():
+        return [
+            'Default',
+            'Human 1']
+
+    @staticmethod
+    def getDefaultOptions(parameterSetName='Default'):
+        options = MeshType_3d_haustra1.getDefaultOptions(parameterSetName)
+        options['Number of elements around'] = 15
+        options['Number of elements along haustrum'] = 4
+        options['Inner radius'] = 1.0
+        options['Haustrum segment mid-derivative factor'] = 2.0
+        options['Wall thickness'] = 0.05
+        optionsColon = {
+            'Number of haustra segments': 30,
+            'Tube type': 2
+            }
+        options.update(optionsColon)
+        if 'Human 1' in parameterSetName:
+            options['Tube type'] = 3
+        return options
+
+    @staticmethod
+    def getOrderedOptionNames():
+        optionNames = MeshType_3d_haustra1.getOrderedOptionNames()
+        optionNames.remove('Haustra segment length')
+        for optionName in [
+            'Number of haustra segments',
+            'Tube type']:
+            optionNames.insert(3, optionName)
+        return optionNames
+
+    def checkOptions(options):
+        MeshType_3d_haustra1.checkOptions(options)
+        if options['Number of haustra segments'] < 1:
+            options['Number of haustra segments'] = 1
+        if options['Tube type'] < 1:
+            options['Tube type'] = 1
+        if options['Tube type'] > 3:
+            options['Tube type'] = 3
+
+    @staticmethod
+    def generateBaseMesh(region, options):
+        """
+        Generate the base tricubic Hermite mesh. See also generateMesh().
+        :param region: Zinc region to define model in. Must be empty.
+        :param options: Dict containing options. See getDefaultOptions().
+        :return: annotationGroups
+        """
+        elementsCountAround = options['Number of elements around']
+        elementsCountAlongHaustrum = options['Number of elements along haustrum']
+        elementsCountThroughWall = options['Number of elements through wall']
+        haustraSegmentCount = options['Number of haustra segments']
+        radius = options['Inner radius']
+        cornerInnerRadiusFactor = options['Corner inner radius factor']
+        haustraInnerRadiusFactor = options['Haustra inner radius factor']
```

Comment:
Carefully update option names and variable names - see comments in 'Haustra 1' scaffold.